test/functional/Jsr292/src/com/ibm/j9/jsr292/FilterArgumentsWithCombinerTest.java

```diff
@@ -0,0 +1,435 @@
+
+        /* verify minimum boundary, filter position is first handle parameter */
+        result = (MethodHandle)fawcMethod.invoke(mhClazz, target, 0, combiner, new int[]{0, 2});
+        AssertJUnit.assertEquals(Short.MIN_VALUE, (short)result.invokeExact(constShort1, Short.MIN_VALUE, constShort2));
+
+        /* verify medium boundary, filter position is last handle parameter */
+        result = (MethodHandle)fawcMethod.invoke(mhClazz, target, 2, combiner, new int[]{1, 2});
+        AssertJUnit.assertEquals(constShort1, (short)result.invokeExact(Short.MAX_VALUE, constShort1, constShort2));
+    }
+
+    /* combiner returns highest short value */
+    public static short combinerShort(short v1, short v2) {
+        return (v1 > v2) ? v1 : v2;
+    }
+
+    /* return the lowest value short */
+    public static short targetShort(short v1, short v2, short v3) {
+       if ((v1 <= v2) & (v1 <= v3)) {
+            return v1;
+        } else if ((v2 <= v1) && (v2 <= v3)) {
+            return v2;
+        } else {
+            return v3;
+        }
+    }
+
+    /* test filterArgumentsWithCombiner for data type char */
+    @Test(groups = { "level.extended" })
+	public void test_filterArgumentsWithCombiner_Character() throws WrongMethodTypeException, Throwable {
+        MethodHandle result;
+
+        /* handle & filter */
+        MethodHandle target = MethodHandles.publicLookup().findStatic(FilterArgumentsWithCombinerTest.class, 
+            "targetChar", MethodType.methodType(char.class, char.class, char.class, char.class));
+        MethodHandle combiner = MethodHandles.publicLookup().findStatic(FilterArgumentsWithCombinerTest.class, 
+            "combinerChar", MethodType.methodType(char.class, char.class, char.class));
+
+        /* verify maximum boundary, filter position is the middle handle parameter */
+        result = (MethodHandle)fawcMethod.invoke(mhClazz, target, 1, combiner, new int[]{2, 0});
+        AssertJUnit.assertEquals(Character.MAX_VALUE, (char)result.invokeExact(Character.MAX_VALUE, constChar2, Character.MAX_VALUE));
+
+        /* verify minimum boundary, filter position is first handle parameter */
+        result = (MethodHandle)fawcMethod.invoke(mhClazz, target, 0, combiner, new int[]{0, 2});
+        AssertJUnit.assertEquals(Character.MIN_VALUE, (char)result.invokeExact(constChar1, Character.MIN_VALUE, constChar2));
+
+        /* verify medium boundary, filter position is last handle parameter */
+        result = (MethodHandle)fawcMethod.invoke(mhClazz, target, 2, combiner, new int[]{1, 2});
+        AssertJUnit.assertEquals(constChar1, (char)result.invokeExact(Character.MAX_VALUE, constChar1, constChar2));
+    }
+
+    /* combiner returns highest char value */
+    public static char combinerChar(char v1, char v2) {
+        return (v1 > v2) ? v1 : v2;
+    }
+
+    /* return the lowest value char */
+    public static char targetChar(char v1, char v2, char v3) {
+       if ((v1 <= v2) & (v1 <= v3)) {
```

Comment:
This can be written in fewer comparisons and ends up with simpler to read logic

```java
public static char targetChar(char v1, char v2, char v3) {
  char lowest = v1;
  if (v2 < lowest) {
   lowest = v2;
  }
  if (v3 < lowest) {
   lowest = v3;
  }
  return lowest;
}
```

Reply:
I've found for tests like this, that sometimes it more clear what is happening if the test either returns a string that concats all the arguments into a list or it returns an array of the arguments.  It helps with debugging the tests when there is a problem.

Reply:
that definitely seems more straightforward. I will do something your suggestion.